src/binding/c/mpit_api.txt

```diff
@@ -0,0 +1,515 @@
+    .skip: validate-TOOL_MPI_OBJ
+{ -- error_check -- pvar_index
+    pvar_table_entry_t *entry;
+    entry = (pvar_table_entry_t *) utarray_eltptr(pvar_table, pvar_index);
+    if (!entry->active) {
+        mpi_errno = MPI_T_ERR_INVALID_INDEX;
+        goto fn_fail;
+    }
+}
+
+MPI_T_pvar_handle_free:
+    .desc: Free an existing handle for a performance variable
+
+MPI_T_pvar_read:
+    .desc: Read the value of a performance variable
+/*
+    Notes:
+    The MPI_T_pvar_read() call queries the value of the performance variable with the
+    handle "handle" in the session identified by the parameter session and stores the result
+    in the buffer identified by the parameter buf. The user is responsible to ensure that the
+    buffer is of the appropriate size to hold the entire value of the performance variable
+    (based on the datatype and count returned by the corresponding previous calls to
+    MPI_T_pvar_get_info() and MPI_T_pvar_handle_alloc(), respectively).
+    
+    The constant MPI_T_PVAR_ALL_HANDLES cannot be used as an argument for the function
+    MPI_T_pvar_read().
+*/
+
+MPI_T_pvar_readreset:
+    .desc: Read the value of a performance variable and then reset it
+
+MPI_T_pvar_reset:
+    .desc: Reset a performance variable
+/*
+    Notes:
+    The MPI_T_pvar_reset() call sets the performance variable with the handle identified
+    by the parameter handle to its starting value. If it is not possible
+    to change the variable, the function returns MPI_T_ERR_PVAR_NO_WRITE.
+    If the constant MPI_T_PVAR_ALL_HANDLES is passed in handle, the MPI implementation
+    attempts to reset all variables within the session identified by the parameter session for
+    which handles have been allocated. In this case, the routine returns MPI_SUCCESS if all
+    variables are reset successfully, otherwise MPI_T_ERR_PVAR_NO_WRITE is returned. Readonly
+    variables are ignored when MPI_T_PVAR_ALL_HANDLES is specified.
+*/
+{
+    MPIR_T_pvar_handle_t *hnd;
+    /* If handle is MPI_T_PVAR_ALL_HANDLES, dispatch the call.
+     * Otherwise, do correctness check, then go to impl.
+     */
+    if (handle == MPI_T_PVAR_ALL_HANDLES) {
+        DL_FOREACH(session->hlist, hnd) {
+            if (!MPIR_T_pvar_is_readonly(hnd)) {
+                mpi_errno = MPIR_T_pvar_reset_impl(session, hnd);
+                if (mpi_errno != MPI_SUCCESS)
+                    goto fn_fail;
+            }
+        }
+    } else {
+        if (handle->session != session) {
+            mpi_errno = MPI_T_ERR_INVALID_HANDLE;
+            goto fn_fail;
+        }
+
+        if (MPIR_T_pvar_is_readonly(handle)) {
+            mpi_errno = MPI_T_ERR_PVAR_NO_WRITE;
+            goto fn_fail;
+        }
+
+        mpi_errno = MPIR_T_pvar_reset_impl(session, handle);
+        if (mpi_errno != MPI_SUCCESS)
+            goto fn_fail;
+    }
+}
+
+MPI_T_pvar_session_create:
+    .desc: Create a new session for accessing performance variables
```

Comment:
missing error `MPI_T_ERR_OUT_OF_SESSIONS`.

Reply:
Again we currently don't really return this error code. Will fix.

Reply:
Added a commit to fix the out of memory behavior.